### PR TITLE
Update/cleanup a bunch of stuff

### PR DESCRIPTION
**Pull request by iggy**
A bunch of interwoven commits (really messy to split them up)

Convert to the more modern style of merging defaults and pillar lookups all in
the map.jinja file.

Split out and clean up logrotate support

Move mongos/files/* under the main files/ dir (there was no reason for them to be separate).

Document mongodb.mongos state's existence

Remove completely invalid init.d startup script

Move mongos/init.sls to mongos.sls (mongos dir is completely empty and gone now

Split tools into it's own state

Update to newer, suggeted repos and add Debian support

### CODE REVIEW

File: mongodb/map.jinja [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
-        'mongos': 'mongos',
-        'conf_path': '/etc/mongodb.conf'
-    },
-    'RedHat': {
-        'pip': 'python-pip',
-        'mongod': 'mongodb',
-        'mongos': 'mongos',
-        'conf_path': '/etc/mongodb.conf'
-    },
-}, merge=salt['pillar.get']('mongodb:lookup')) %}
```
Comment (babilen) [inline mongodb/map.jinja]: Aren't you getting rid of the possibility to override things via `mongodb:lookup` here, which renders this backward incompatible?

Comment (iggy) [inline mongodb/map.jinja]: I can add that in, but honestly, I highly doubt anyone uses this formula as is. It is impossible to get much beyond a single host deployment up with what's here. And in that case, you could have just as easily just installed the package and used the default config.

Comment (babilen) [inline mongodb/map.jinja]: I kind of like keeping `:lookup:` for overrides in `os_map` around with the added benefit of backwards compatibility in this case.

Comment (iggy) [inline mongodb/map.jinja]: Done

**Comment by gravyboat** [comment]
This looks good so I'm going to merge it. If anyone has any further concerns please discuss them here. Thanks @iggy!

**Comment by iggy** [comment]
I've also got some more, bigger changes coming. Hopefully easy setup of sharded and replicated clusters. More flexible data layout. More config options supported in the config file templates. Mongos and mongodb not stomping all over each other when running on the same host. And much much more!

**Pull request merged**